### ISSUE TITLE
Bounding Box not scrolling during marquee selection of elements beyond visible area

### ISSUE THREAD

**Issue by idleup** [smeighan/xLights#638]
If your sequence has enough rows that you have a vertical and or horizontal scroll bar and you left click and drag to create a bounding box, the page does will not scroll down when you drag the bounding box above the top row, below the bottom visible row, or to the sides. If you attempt to use the mouse wheel to scroll while holding the left mouse button to maintain your bounding box than the whole bounding box moves with the paging and it looses what was selected before the scroll. 

Basically, you can only select elements using a bounding box on the visible part of the screen when you start the bounding box. 

Unless this is an easy fix it probably is not a high priority as you can always just control-click every element you want selected, scroll the page, select more to add to your selection group, repeat, etc until you have all your elements selected. It just takes longer.

**Comment by AzGilrock** [comment]
Does the bounding box stay in place if you don't try to keep the left mouse button down?  I though it did.  If so then create a box, scroll down and try holding down Shift and then left click.  This works when you are using timing cells but I can't remember if it works in freeform selection mode.

**Comment by idleup** [comment]
I can't get that to work in timing or freeform selection mode, both ways it deselects what was in the first bounding box prior to scrolling to the second one

**Comment by AzGilrock** [comment]
What action is causing a deselect?  If you have a block of cells selected you should be able to change the size of that block selection by holding down shift and clicking in another cell.  Try it without scrolling first to see it work.

**Comment by AzGilrock** [comment]
Ok I see what it's doing but its not easy to fix.  I kinda hate our design in this area.  None of the rows have fixed numbers every time you expand/contract strands and nodes the row numbers are completely recalculated.  There is also all this VisibleRowInformation versus total RowInformation.  Then you factor the two different selection modes and that model rows can scroll up under timing rows and it gets crasy to keep track of it all.  I feel like it's a big Jenga game and I move one block and it all comes crashing down.

**Comment by idleup** [comment]
Thanks!
